src/extension/src/azure/azureServices.ts

```diff
@@ -287,26 +287,66 @@ export class AzureServices extends WizardServant {
     }
   }
 
-  public static async deployResourceGroup(payload: any): Promise<any> {
+  public static async generateDistinctResourceGroupSelections(
+    payload: any
+  ): Promise<ResourceGroupSelection[]> {
     const projectName = payload.engine.projectName;
-    const functionSubscription = payload.functions.subscription;
+    let result: ResourceGroupSelection[] = [];
+    let allSubscriptions: SubscriptionItem[] = [];
 
-    await AzureServices.updateFunctionSubscriptionItemCache(
-      functionSubscription
-    );
+    if (AzureServices.functionsSelectedNewResourceGroup(payload)) {
+      await AzureServices.updateFunctionSubscriptionItemCache(
+        payload.functions.subscription
+      );
+      allSubscriptions.push(AzureServices.usersFunctionSubscriptionItemCache);
+    }
+    if (AzureServices.cosmosDBSelectedNewResourceGroup(payload)) {
+      await AzureServices.updateCosmosDBSubscriptionItemCache(
+        payload.cosmos.subscription
+      );
+      allSubscriptions.push(AzureServices.usersCosmosDBSubscriptionItemCache);
+    }
+    const allDistinctSubscriptions: SubscriptionItem[] = [... new Set(allSubscriptions)];
+    const generatedName: string = await AzureServices.AzureResourceGroupProvider.generateValidResourceGroupName(
+          projectName,
+          allDistinctSubscriptions
+        );
+    let tempResourceGroupSelection: ResourceGroupSelection;
+    allDistinctSubscriptions.forEach(subscription => {
+      tempResourceGroupSelection = AzureServices.generateResourceGroupSelection(
+        generatedName,
+        subscription
+      );
+      result.push(tempResourceGroupSelection);
+    });
 
-    const generatedName = await AzureServices.AzureResourceGroupProvider.generateValidResourceGroupName(
-      projectName,
-      this.usersFunctionSubscriptionItemCache
-    );
+    return result;
+  }
 
-    const resourceGroupSelection: ResourceGroupSelection = {
-      subscriptionItem: AzureServices.usersFunctionSubscriptionItemCache,
+  public static functionsSelectedNewResourceGroup(payload: any): boolean {
```

Comment:
is the payload really any here? Does it need to be public?

Reply:
The payload is what the client sends to the extension.
Yes, it is used in the generateExperience too, so it has to be public.

Reply:
if its used in generateExperience never mind